pkg/build/builder/common.go

```diff
@@ -201,15 +217,24 @@ func retryBuildStatusUpdate(build *buildapi.Build, client client.BuildInterface,
 		latestBuild.Status.Output.To = build.Status.Output.To
 		latestBuild.Status.Stages = build.Status.Stages
 
-		if _, err := client.UpdateDetails(latestBuild); err != nil {
-			return err
+		_, err = client.UpdateDetails(latestBuild)
+		netErr, ok := err.(net.Error)
+
+		switch {
+		case err == nil:
+			return true, nil
+		case errors.IsConflict(err):
+			latestBuild = nil
+			return false, nil
+		case ok && netErr.Temporary():
+			return false, nil
 		}
-		return nil
+
+		return false, err
+
 	})
-}
 
-func handleBuildStatusUpdate(build *buildapi.Build, client client.BuildInterface, sourceRev *buildapi.SourceRevision) {
-	if updateErr := retryBuildStatusUpdate(build, client, sourceRev); updateErr != nil {
+	if updateErr != nil {
 		glog.Infof("error: Unable to update build status: %v", updateErr)
```

Comment:
i'm guessing this is always going to be a timeout error (because that's what ExponentialBackoff will throw once it gives up).  two suggestions:

1) store the error value you're getting within the retry func, so that you can report at least the last error it encountered, here.
2) add some logging in the retry func so when we are getting errors and retrying, we can see that in the logs.

Otherwise we're just going to see that the retry ultimately failed due to a retry timeout, but we won't know why it had to retry.